cdxev/initialize_sbom.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import json
+import warnings
+from datetime import datetime
+from typing import Any, Union
+
+from cyclonedx.model import (  # type: ignore
+    ExternalReference,
+    ExternalReferenceType,
+    Tool,
+    XsUri,
+)
+from cyclonedx.model.bom import Bom, BomMetaData  # type: ignore
+from cyclonedx.model.bom_ref import BomRef  # type: ignore
+from cyclonedx.model.component import Component, ComponentType  # type: ignore
+from cyclonedx.model.contact import (  # type: ignore
+    OrganizationalContact,
+    OrganizationalEntity,
+)
+from cyclonedx.model.dependency import Dependency  # type: ignore
+from cyclonedx.output.json import JsonV1Dot6  # type: ignore
+
+from cdxev import pkg
+
+
+def initialize_sbom(
+    software_name: Union[str, None],
+    version: Union[str, None],
+    supplier_name: Union[str, None],
+) -> dict[str, Any]:
+    """
+    Creates a initial SBOM draft to work with, containing the most basic fields.
+
+    param software_name: the name of the software the sbom is for
+    param version: the version of the sbom the software is for
+    param supplier_name: name of the supplier the software is from 
```

Comment:
see comment in main regarding this variable being used in `metadata.supplier` too.

Reply:
New variable added